File: WdtcCore/src/main/java/org/wdt/wdtc/core/download/source/OfficialDownloadSource.java
package org.wdt.wdtc.core.download.source;

import org.wdt.wdtc.core.download.infterface.DownloadSourceInterface;

public class OfficialDownloadSource implements DownloadSourceInterface {
    @Override
    public String getLibraryUrl() {
        return "https://libraries.minecraft.net/";
    }

    @Override
    public String getAssetsUrl() {
        return "https://resources.download.minecraft.net/";
    }

    @Override
    public String getMetaUrl() {
        return "https://piston-meta.mojang.com/";
    }

    @Override
    public String getDataUrl() {
        return "https://piston-data.mojang.com/";
    }

    @Override
    public String getVersionManifestUrl() {
        return "https://piston-meta.mojang.com/mc/game/version_manifest.json";
    }

    @Override
    public String getFabricMetaUrl() {
        return "https://meta.fabricmc.net/";
    }

    @Override
    public String getFabricLibraryUrl() {
        return "https://maven.fabricmc.net/";
    }

    @Override
    public String getForgeLibraryMavenUrl() {
        return "https://files.minecraftforge.net/maven/";
    }

    @Override
    public String getVersionClientUrl() {
        throw new RuntimeException("Official doesn't have version client url");
    }

    @Override
    public String getOfficialUrl() {
        return getMetaUrl();
    }


}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/download/source/OfficialDownloadSource.kt
package org.wdt.wdtc.core.download.source

import org.wdt.wdtc.core.download.infterface.DownloadSourceInterface

class OfficialDownloadSource : DownloadSourceInterface {
    override val libraryUrl: String
        get() = "https://libraries.minecraft.net/"
    override val assetsUrl: String
        get() = "https://resources.download.minecraft.net/"
    override val metaUrl: String
        get() = "https://piston-meta.mojang.com/"
    override val dataUrl: String
        get() = "https://piston-data.mojang.com/"
    override val versionManifestUrl: String
        get() = "https://piston-meta.mojang.com/mc/game/version_manifest.json"
    override val fabricMetaUrl: String
        get() = "https://meta.fabricmc.net/"
    override val fabricLibraryUrl: String
        get() = "https://maven.fabricmc.net/"
    override val forgeLibraryMavenUrl: String
        get() = "https://files.minecraftforge.net/maven/"
    override val versionClientUrl: String
        get() {
            throw RuntimeException("Official doesn't have version client url")
        }
    override val officialUrl: String
        get() = metaUrl
}
